src/runtime/types.h

```diff
@@ -435,14 +455,63 @@ class BoxedList : public Box {
 class BoxedTuple : public Box {
 public:
     typedef std::vector<Box*, StlCompatAllocator<Box*>> GCVector;
-    GCVector elts;
 
-    BoxedTuple(GCVector& elts) __attribute__((visibility("default"))) : elts(elts) {}
-    BoxedTuple(GCVector&& elts) __attribute__((visibility("default"))) : elts(std::move(elts)) {}
+    Box** elts;
+
+    void* operator new(size_t size, size_t nelts) __attribute__((visibility("default"))) {
+        Box* rtn = static_cast<Box*>(gc_alloc(_PyObject_VAR_SIZE(tuple_cls, nelts + 1), gc::GCKind::PYTHON));
+        rtn->cls = tuple_cls;
+        return rtn;
+    }
+
+    void* operator new(size_t size, BoxedClass* cls, size_t nelts) __attribute__((visibility("default"))) {
+        Box* rtn = static_cast<Box*>(cls->tp_alloc(cls, nelts));
+        rtn->cls = cls;
+        return rtn;
+    }
+
+    static BoxedTuple* Create(int64_t size) { return new (size) BoxedTuple(size); }
```

Comment:
We've usually been using camelCase for function names (ie this would be `create`).